server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java

```diff
@@ -1218,11 +1219,178 @@ void createRetentionLease(long startingSeqNo, ActionListener<RetentionLease> lis
         }
     }
 
+    public void testSeqNoBasedRecoveryCleanSnapshotFilesAndRecoversFromFallbackPlanAfterAFailure() throws Exception {
```

Comment:
I am not sure I understand the name here, at least the `CleanSnapshotFiles` is not not clear. Perhaps remove that part of the name?